astropy/units/tests/test_quantity.py

```diff
@@ -1003,6 +1003,169 @@ def test_to_string(self):
+            (1234567, "m", "_", "1_234_567.0 m", None, None),
+            (u.Quantity(2.5 - 1.2j), None, ".2f", "2.50-1.20j", None, None),
+            (u.Quantity(2.5 - 1.2j), None, ".2e", "2.50e+00-1.20e+00j", None, None),
+            # +, - and space are ignored in complex formatting
+            (
+                u.Quantity(2.5 - 1.2j),
+                None,
+                "+.2f",
+                r"$(2.50-1.20i) \; \mathrm{}$",
+                "latex",
+                None,
+            ),
+            (
+                u.Quantity(2.5 - 1.2j),
+                None,
+                "-.2f",
+                r"$(2.50-1.20i) \; \mathrm{}$",
+                "latex",
+                None,
+            ),
+            (
+                u.Quantity(2.5 - 1.2j),
+                None,
+                ">+.2f",
+                r"$(2.50-1.20i) \; \mathrm{}$",
+                "latex",
+                None,
+            ),
+            (
+                u.Quantity(2.5 - 1.2j),
+                None,
+                " .2f",
+                r"$(2.50-1.20i) \; \mathrm{}$",
+                "latex",
+                None,
+            ),
+            (
+                137000000,
+                "lyr",
+                ">+30,.2e",
+                "                     +1.37e+08 lyr",
+                None,
+                None,
+            ),
+            (
+                1.23456789e15,
+                "m/s",
+                ".3e",
+                r"$1.235 \times 10^{15} \; \mathrm{\frac{m}{s}}$",
+                "latex",
+                None,
+            ),
+            (
+                123.456,
+                "km/s",
+                ".2f",
+                r"$123.46 \; \mathrm{\frac{km}{s}}$",
+                "latex",
+                None,
+            ),
+            (
+                123.456,
+                "m/s",
+                ".2f",
+                r"$123.46 \; \mathrm{m\,s^{-1}}$",
+                "latex_inline",
+                None,
+            ),
+            (
+                123.456,
+                "m/s",
+                ".3e",
+                r"$1.235 \times 10^{2} \; \mathrm{\frac{m}{s}}$",
+                "latex",
+                None,
+            ),
+            (
+                123.456,
+                "m/s",
+                ".3e",
+                r"$1.235 \times 10^{2} \; \mathrm{m\,s^{-1}}$",
+                "latex_inline",
+                None,
+            ),
+        ],
+    )
+    def test_format_spec(
+        self, input_value, input_unit, format_spec, expected_result, format, precision
+    ):
+        qscalar = u.Quantity(input_value, input_unit)
+        assert (
+            qscalar.to_string(formatter=format_spec, precision=precision, format=format)
+            == expected_result
+        )
+
+    @pytest.mark.parametrize(
+        "input_quantity, formatter, format, expected_result",
+        [
+            # More explicit formatting
```

Comment:
Use [`pytest.param()` with a custom `id`](https://docs.pytest.org/en/stable/example/parametrize.html#set-marks-or-test-id-for-individual-parametrized-test) instead of writing out these comments.